lib/views/setting/setting_view.dart

```diff
@@ -126,63 +120,155 @@ class _SettingViewState extends State<SettingView> {
+    }
+  }
+
+  Widget _buildMainView({required BuildContext context}) {
+    return SingleChildScrollView(
+      child: Padding(
+        padding: EdgeInsets.only(
+            left: MediaQuery.of(context).size.width * 0.2,
+            right: MediaQuery.of(context).size.width * 0.2),
+        child: Wrap(
+          runSpacing: MediaQuery.of(context).size.height * 0.05,
+          children: [
+            const Text("Connection type"),
+            InputDecorator(
+              decoration: const InputDecoration(border: OutlineInputBorder()),
+              child: DropdownButton(
+                value: setting.clientMode,
+                underline: const SizedBox(),
+                items: clients.map((ClientMode items) {
+                  return DropdownMenuItem(
+                    value: items,
+                    child: Text(items.toString()),
+                  );
+                }).toList(),
+                onChanged: (ClientMode? newValue) {
+                  setState(() {
+                    setting.clientMode = newValue!;
+                  });
+                },
+              ),
+            ),
+            const Text("Node Nick Name "),
+            TextFormField(
+              controller: nickNameController,
+              onChanged: (name) {
+                setState(() {
+                  setting.nickName = name;
+                });
+              },
+              decoration: const InputDecoration(
+                border: OutlineInputBorder(),
+                hintText: 'My lightning node',
+              ),
+            ),
+            _buildCorrectSettingView(context: context, setting: setting),
+            Row(
+              children: [
+                ElevatedButton(
+                  onPressed: () {
+                    if (setting.isValid()) {
+                      saveSettings();
+                      RegisterProvider
+                          .registerClientFromSetting(setting, widget.provider);
+                      Navigator.of(context).pushAndRemoveUntil(
+                          MaterialPageRoute(
+                              builder: (context) => HomeView(
+                                    provider: widget.provider,
+                                  )),
+                          (Route<dynamic> route) => false);
+                    }
+                  },
+                  child: const Text("Save"),
+                ),
+                const Spacer(),
+                ElevatedButton(
+                  onPressed: () async {
+                    final prefs = await SharedPreferences.getInstance();
+                    prefs.remove("setting");
+                    getSettingsInfo().then((value) => {
+                          setState(() {
+                            setting = value;
+                            nickNameController.clear();
+                            hostController.text = "localhost";
+                          }),
+                        });
+                  },
```

Comment:
I preferer to put this inside a utils function, and also you forget to unregister the ApiApp inside the Provider.

You can add a method to unregister it